app/controllers/users/sessions_controller.rb

```diff
@@ -39,6 +40,13 @@ def timeout
       redirect_to root_url
     end
 
+    protected
+
+    def require_no_authentication
```

Comment:
maybe instead of calling `super` we should rename this and add it as a `before_action` for `new` ? I think that might be more readable. I am also a tad confused about why we are adding this and skipping it for certain actions. Is it called by default in `Devise::SessionsController`, perhaps?